### DISCUSSION TITLE
Weberbox PCB troubleshooting

### DISCUSSION THREAD

**Discussion by acw380** [nebhead/PiFire#45] [Q&A] [answered]
I am currently in the process of finishing my build but I am having some issues that I am hoping I can get some help troubleshooting. I ordered the weberbox pcb from JLCPCB and soldered up all the components from the original Pifire post. The auto-installer for the code worked fine but It seem to be having issues running any of the grill components. I am just getting all zeros on the temp gauges and I am not getting any voltage sent to my relay board.  I have gone through my circuit and the only issue I have seen is that it looks like one pin on the raspberry pi is different on weberbox pcb compared to the original post from nebhead. Did you have to change the code any for your board to work weberbox? 
![image](https://user-images.githubusercontent.com/91814900/135895470-8ac0594c-d4d9-4814-ba87-822471f75aec.png)

**Comment by weberbox** [comment, marked as answer]
I believe that image you referenced is incorrect. According to the PCB specs CE0 should be GPIO8 (pin24). The PCB I designed and the original from nebhead have the same GPIO pinout. There should not be any software changes needed to get the PCB I designed to work. My design has a fourth probe input and that is currently not supported by nebhead's branches.  (The other three probes will work fine, Grill1, Probe1 and Probe2)

The probe temps will only show if PiFire is in a work mode or monitor mode. 

What relay PCB are you using? There are two configurations either being pull up or pull down. Which is set via settings.json under globals triggerlevel. The default is LOW or pull down

```
settings['globals'] = {
		'grill_name' : '',
		'debug_mode' : False,
		'page_theme' : 'light',
		'triggerlevel' : 'LOW',
		'buttonslevel' : 'HIGH',
		'shutdown_timer' : 60,
		'units' : 'F'
	}
```

![Screenshot 2021-10-04 095422](https://user-images.githubusercontent.com/313328/135902927-41fed514-ae85-467e-b6a5-20de2292c63c.png)

![Screenshot 2021-10-04 094718](https://user-images.githubusercontent.com/313328/135902935-f3fc7196-3b84-4606-920e-966f095a621d.png)

**Reply by nebhead** [reply]
> The probe temps will only show if PiFire is in a work mode or monitor mode.

^^ Came here to say this ^^ Definitely try pressing the play button on the WebUI first to start the control script running.  This should start indicating temperatures for probes connected, as well as toggling the relays for fan, ignitor, auger.  
 
> What relay PCB are you using? There are two configurations either being pull up or pull down. Which is set via settings.json under globals triggerlevel. The default is LOW or pull down

Also important - but can't recall if we made the change to the main branch or just the development branch.  I think we are probably ready to merge the development back into main, but before I do that I just wanted to make sure that the install script is running properly and was thinking to include the triggerlevel globals as a configurable item in the installation process.

**Reply by acw380** [reply]
Thanks for the quick responses. 

Weberbox, yes that is how my board is wired, thanks. I think I had a few issues on my end but I have it functioning now. Here is the relay I bought ([link](https://www.amazon.com/gp/product/B07DK6P9CV/ref=ppx_yo_dt_b_asin_title_o04_s01?ie=UTF8&psc=1)) from the recommendations page on the pifire page. It is the low level trigger solid state relay. I had re-flashed/installed the code onto my pi three times over the weekend but this last time I did not add any of the optional items to the config.txt file and it worked this time. I think I must have been adding it in the wrong place. I know most people on here probably wouldnt mess that part up but I am a bit new to this sort of project. Should the end of my config file look like this?

[pi4]
# Enable DRM VC4 V3D driver on top of the dispmanx display stack
dtoverlay=vc4-fkms-v3d
max_framebuffers=2

[all]
#dtoverlay=vc4-fkms-v3d
#dtparam=i2c_arm=on,i2c_baudrate=400000
#core_freq=250

dtparam=spi=on

**Reply by weberbox** [reply]
I do not remember exactly what mine looks like but you should have the below items in the config.txt

```
dtparam=i2c_arm=on,i2c_baudrate=400000
core_freq=250
dtparam=spi=on
```

`dtparam=spi=on` is if you are using a ili9341 display.

Only thing I noticed is a couple of your lines are commented out with a `#` if you want them active you must remove the `#`

**Reply by nebhead** [reply]
BTW... I'm currently running **without** the I2C bus speed enhancement turned on.  I did feel like this was necessary when I was running the display on I2C, alongside the ADS1115 ADC, but since using the SPI interface for the display, it doesn't really seem like the extra speed is needed.  

And as mentioned, if you are using low level trigger relays (as it appears you are), then you'll want to make sure your triggerlevel is set to low in the settings.json file. And if you are using the buttons with the display, you'll want to make sure you have the buttonslevel set correctly too. 

```
	settings['globals'] = {
		'grill_name' : '',
		'debug_mode' : False,
		'page_theme' : 'light',
		'triggerlevel' : 'LOW',
		'buttonslevel' : 'HIGH',
		'shutdown_timer' : 60,
		'units' : 'F'
	}
```

**Reply by acw380** [reply]
Thanks for your help. I got my grill up and running and its working great. Its a bit variable but I think my smoker must have a base issue with the auger or something. Before I built this controller I was getting crazy swings (common 50 to 75 over temp) on the traegar controller I had and the after market Ortech controller I bought. With this Im getting a much more reasonable +- 10 to 15 or so with the occasional 20-30 swing (over my 5 hour test). Is there any thoughts on having built in troubleshooting on grill components? I did notice that there was a flameout warning that popped up which was nice to see. Just curious if there could be a something that would log unexpected response times or something? My build isnt the nicest looking but I think Ill post some pictures to your build thread to add another take on it.

**Reply by nebhead** [reply]
> Thanks for your help. I got my grill up and running and its working great. Its a bit variable but I think my smoker must have a base issue with the auger or something. Before I built this controller I was getting crazy swings (common 50 to 75 over temp) on the traegar controller I had and the after market Ortech controller I bought. With this Im getting a much more reasonable +- 10 to 15 or so with the occasional 20-30 swing (over my 5 hour test). 

There will always be some variability given the nature of the system.  But the PID can be tuned to improve things.  That said, I'm not a PID expert and I'm not really even sure how to approach that.  I've noticed that on mine, I can get a nice +-5-10F delta consistantly, if I'm setting the hold above 200F.  I'll have to post some results after my next cook.  However, there is still a problem with ramping up too high in the very beginning, and with compensation when the lid is open.  If I never open the lid, the PID does a really nice job honing in on the temperature after 20 minutes or so.  

> Is there any thoughts on having built in troubleshooting on grill components? I did notice that there was a flameout warning that popped up which was nice to see. Just curious if there could be a something that would log unexpected response times or something? My build isnt the nicest looking but I think Ill post some pictures to your build thread to add another take on it.

Yes, I recently built in a time-out into the pellet sensor (it's in the development branch right now) which will detect if it's taken too long to check the pellets.  I find this happens occasionally and it sort of stalls the entire grill.   I'm thinking about making this event a thread.  We could do the same with the ADC and the display.  As for the grill components (igniter, auger, fan) it's tough to get feedback from those components since they are isolated from the system.  The best we have may be the safety features to alert of flameout or too high temp.  

Would be great to see pictures of your build!

**Reply by weberbox** [reply]
>However, there is still a problem with ramping up too high in the very beginning, and with compensation when the lid is open.  If I never open the lid, the PID does a really nice job honing in on the temperature after 20 minutes or so.  
> 

I have not looked at the PID code much but I wonder if it would be possible to implement something like the new smoke plus change. 

Maybe alter the pellet feed rate until the grill reaches or gets near the set temp, then kick the PID into action. I don't know if this would cause the PID to freak out but it seems like that is what is happening now anyway. The PID sees the temp way below and just feeds so many pellets without really pausing.
Then it does not slow down when it nears the set temp so the pot just has a ton of pellets to burn and the temp shoots way over. 

I will take a look when I have some time and rack my brain a bit.
